src/github/pull-request.ts

```diff
@@ -1,61 +1,60 @@
 import { transformPullRequest } from "../transforms/pull-request";
 import issueKeyParser from "jira-issue-key-parser";
-
 import { emitWebhookProcessedMetrics } from "../util/webhooks";
 import { CustomContext } from "./middleware";
-import _ from "lodash";
-
-export default async (
-	context: CustomContext,
-	jiraClient,
-	util
-): Promise<void> => {
+import { isEmpty } from "lodash";
+import { booleanFlag, BooleanFlags } from "../config/feature-flags";
+import GitHubClient from "./client/github-client";
+import { getCloudInstallationId } from "./client/installation-id";
+import { GitHubAPI } from "probot";
+import { Octokit } from "@octokit/rest";
+
+export const pullRequestWebhookHandler = async (context: CustomContext, jiraClient, util, githubInstallationId: number): Promise<void> => {
 	const {
 		pull_request,
 		repository: {
 			id: repositoryId,
 			name: repo,
-			owner: { login: owner },
+			owner: { login: owner }
 		},
-		changes,
+		changes
 	} = context.payload;
+	const { number: pullRequestNumber, body: pullRequestBody, id: pullRequestId } = pull_request;
+	const baseUrl = jiraClient.baseUrl || "none";
+	const githubClient =
+		await booleanFlag(BooleanFlags.USE_NEW_GITHUB_CLIENT_FOR_PULL_REQUEST_WEBHOOK, false, jiraClient.baseURL) ?
+			new GitHubClient(getCloudInstallationId(githubInstallationId), context.log)
+			: context.github;
+
+	context.log = context.log.child({
+		number: pullRequestNumber,
+		body: pullRequestBody,
+		id: pullRequestId
```

Comment:
the "id" parameter is already used for sqs message id.
We need to choose other names.

Also, we shoudln't log PR body, because it is UGC (Only log in case of errors)
- [ ] Adjust logger parameters

Reply:
We are more interested at adding jiraHostname and InstallationID to logger params

Reply:
Have dropped the body....

and used more descriptive names for the log attributes, squeezed in install and jirahost too
```
{
 jiraHostName: jiraClient.baseURL,
 installationId: githubInstallationId,
 pullRequestNumber,
 pullRequestId
}
```